march_gazebo_plugins/src/com_controller.cpp

```diff
@@ -0,0 +1,210 @@
+#include <gazebo/common/common.hh>
+#include <gazebo/physics/physics.hh>
+#include <march_shared_resources/GaitActionGoal.h>
+#include <ros/callback_queue.h>
+#include <ros/ros.h>
+
+namespace gazebo
+{
+class ComController : public ModelPlugin
+{
+public:
+  void Load(physics::ModelPtr _parent, sdf::ElementPtr /*_sdf*/)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load "
+                       "plugin. "
+                       << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in "
+                          "the gazebo_ros package)");
+      return;
+    }
+
+    // Initialise variables
+    this->model = _parent;
+    this->foot_left = this->model->GetLink("ankle_plate_left");
+    this->foot_right = this->model->GetLink("ankle_plate_right");
+
+    this->swing_step_size = 0.7;  // This estimate will be adjusted every step
+    this->subgait_name = "home_stand";
+
+    this->p_pitch = 60;
+    this->d_pitch = 10;
+    this->p_roll = 150;
+    this->d_roll = 7.5;
+    this->p_yaw = 500;
+    this->d_yaw = 25;
+
+    this->error_x_last_timestep = 0;
+    this->error_y_last_timestep = 0;
+    this->error_yaw_last_timestep = 0;
+
+    // Listen to the update event. This event is broadcast every
+    // simulation iteration.
+    this->updateConnection = event::Events::ConnectWorldUpdateBegin(std::bind(&ComController::OnUpdate, this));
+
+    // Create our ROS node.
+    this->rosNode.reset(new ros::NodeHandle("gazebo_client"));
```

Comment:
You should use `std::make_unique` which is [more modern](https://clang.llvm.org/extra/clang-tidy/checks/modernize-make-unique.html)
```suggestion
    this->rosNode = std::make_unique<ros::NodeHandle>(ros::NodeHandle("gazebo_client"));
```